### PR TITLE
Add Article and Methodology page types

### PR DESCRIPTION
**Pull request by zerolab**
[NWP-258](https://jira.ons.gov.uk/browse/NWP-258) and [NWP-281](https://jira.ons.gov.uk/browse/NWP-281)

* Adds a methodologies app with methodology page model and separate fields for the background data. Note that I chose a simple output in the template (rather than the table)
* Adda an articles app with: article series and article page models
* Moves the bulletin `StreamBlock`s to core app 
* Moves the topic <-> page relation to the taxonomy app

Next step is to extract the common elements to abstract models for DRY, but IMHO this is enough for this POC

### CODE REVIEW

File: ons_alpha/jinja2/templates/pages/article_page.html [hunk 1]
```diff
@@ -0,0 +1 @@
+{% extends "templates/bulletin_page.html" %}
```
Comment (MebinAbraham) [inline ons_alpha/jinja2/templates/pages/article_page.html]: This reference results in an error.
```suggestion
{% extends "templates/pages/bulletin_page.html" %}
```

File: ons_alpha/articles/models.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+class ArticlePage(BundledPageMixin, RoutablePageMixin, BasePage):
+    base_form_class = PageWithUpdatesAdminForm
+    template = "templates/pages/article_page.html"
+    parent_page_types = ["ArticleSeriesPage"]
+    subpage_types = []
+
+    summary = models.TextField()
+    release_date = models.DateField()
+    next_release_date = models.DateField()
```
Comment (MebinAbraham) [inline ons_alpha/articles/models.py:35]: Not a blocker. Articles don't seem to display a "next release date".

File: ons_alpha/jinja2/templates/pages/article_page.html (hunk 1 above)
Comment (zerolab) [reply ons_alpha/jinja2/templates/pages/article_page.html]: gah, overzealous last minute cleanup. great catch. sorted

File: ons_alpha/methodologies/models.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+class MethodologyPage(BundledPageMixin, BasePage):
+    base_form_class = PageWithTopicsAdminForm
+    template = "templates/pages/methodology_page.html"
+    parent_page_types = ["topics.TopicPage"]
+    subpage_types = []
+
+    summary = models.TextField()
+    last_revised_date = models.DateField(blank=True, null=True)
+    how_it_was_revised = models.CharField(blank=True, max_length=255)
```
Comment (MebinAbraham) [inline ons_alpha/methodologies/models.py]: Where this is rendered looks odd but okay for the poc.

File: ons_alpha/articles/models.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+        # tweak the admin display title to include the parent title
+        return f"{self.get_parent().title}: {self.draft_title or self.title}"
+
+    def get_context(self, request, *args, **kwargs):
+        context = super().get_context(request, *args, **kwargs)
+        context["toc"] = self.toc
+        return context
+
+    @path("previous/v<int:version>/")
+    def previous_version(self, request, version):
```
Comment (RealOrangeOne) [inline ons_alpha/articles/models.py:129]: Suggestion: Much of this updates code is duplicated from bulletins. Perhaps it should live in a mixin?

File: ons_alpha/core/blocks/markup.py [hunk 5]
```diff
@@ -12,9 +12,11 @@ class Meta:
         icon = "title"
         form_classname = "title"
         template = "templates/components/streamfield/heading_block.html"
+        label = "Section heading"
 
     def __init__(self, **kwargs):
         self.show_back_to_toc = kwargs.pop("show_back_to_toc", False)
+        kwargs["help_text"] = kwargs.pop("help_text", "This is output as level 2 heading (<code>h2</code>)")
```
Comment (RealOrangeOne) [inline ons_alpha/core/blocks/markup.py]:
```suggestion
        kwargs.setdefault("help_text", "This is output as level 2 heading (<code>h2</code>)")
```

**Review by MebinAbraham — COMMENTED** [review]
Articles/Methodology doesn't seem to add to a bundle correctly. If you click "Add to bundle" from Article/Methodology page, then go to the bundle, you can see it has tried to link it but seems invalid. If you click Save on the bundle, the pages get removed from the bundle.

File: ons_alpha/jinja2/templates/pages/methodology_page.html [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                <img src="https://cdn.ons.gov.uk/assets/images/ons-logo/kitemark/v2/uksa-kitemark-en.svg" alt="UK Statistics Authority Kitemark" class="osr__logo" width="90">
+            </a>
+        {% endif %}
+    </h1>
+
+    <div class="ons-grid ons-u-ml-no">
+        {% if page.last_revised_date %}
+            <div class="ons-grid__col ons-grid__col--sticky@m ons-col-4@m ons-col-12@s ons-u-p-no">
+                <span class="ons-u-fs-r--b">{{ _("Last revised") }}:</span>
+                <span class="ons-u-nowrap">{{ page.last_revised_date|date("j F Y") }}</span>
```
Comment (RealOrangeOne) [inline ons_alpha/jinja2/templates/pages/methodology_page.html:16]: Suggestion: Named formats should be used to be consistent across the site.

File: ons_alpha/articles/models.py (hunk 2 above)
Comment (zerolab) [reply ons_alpha/articles/models.py:35]: it works on my machine?

Screenshot 2024-07-26 at 09 30 56

File: ons_alpha/methodologies/models.py (hunk 3 above)
Comment (zerolab) [reply ons_alpha/methodologies/models.py]: I went back and forth on where to display it. Will move it to the main content area

File: ons_alpha/jinja2/templates/pages/methodology_page.html [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+            </div>
+        {% endif %}
+        {% if page.how_it_was_revised %}
+            <div class="ons-grid__col ons-grid__col--sticky@m ons-col-4@m ons-col-12@s ons-u-p-no">
+                <span class="ons-u-fs-r--b">{{ _("How it was compiled") }}:</span>
+                <span class="ons-u-nowrap">{{ page.how_it_was_revised }}</span>
```
Comment (RealOrangeOne) [inline ons_alpha/jinja2/templates/pages/methodology_page.html]: Question: Is it intentional that there's a difference between "compiled" and "revised" here?

File: ons_alpha/articles/models.py (hunk 2 above)
Comment (MebinAbraham) [reply ons_alpha/articles/models.py:35]: Sorry I wasn't clear. I mean I couldn't see this field on the live website. Have you got an example?

File: ons_alpha/articles/models.py (hunk 4 above)
Comment (zerolab) [reply ons_alpha/articles/models.py:129]: the PR description explicitly calls this out.

File: ons_alpha/jinja2/templates/pages/methodology_page.html (hunk 6 above)
Comment (zerolab) [reply ons_alpha/jinja2/templates/pages/methodology_page.html:16]: one for another PR

File: ons_alpha/jinja2/templates/pages/methodology_page.html (hunk 7 above)
Comment (zerolab) [reply ons_alpha/jinja2/templates/pages/methodology_page.html]: naming thing and all that...

File: ons_alpha/articles/models.py (hunk 2 above)
Comment (zerolab) [reply ons_alpha/articles/models.py:35]: https://www.ons.gov.uk/economy/governmentpublicsectorandtaxes/publicsectorfinance/articles/developmentofpublicsectorfinancestatistics/may2024

**Comment by zerolab** [comment]
@MebinAbraham 
> Articles/Methodology doesn't seem to add to a bundle correctly. If you click "Add to bundle" from Article/Methodology page, then go to the bundle, you can see it has tried to link it but seems invalid. If you click Save on the bundle, the pages get removed from the bundle.

Sorted in 3ed79f4

**Review by MebinAbraham — APPROVED** [review]

**Pull request merged**